tokio/src/time/driver/tests/test_inner.rs

```diff
@@ -0,0 +1,21 @@
+#![warn(rust_2018_idioms)]
+
+use crate::park::Unpark;
+use crate::time::driver::Inner;
+use crate::time::Instant;
+
+struct MockUnpark;
+
+impl Unpark for MockUnpark {
+    fn unpark(&self) {}
+}
+
+#[test]
+fn sanity() {
+    let unpark = Box::new(MockUnpark);
+    let instant = Instant::now();
+
+    let inner = Inner::new(instant, unpark);
+
+    assert_eq!(inner.elapsed(), 0);
```

Comment:
I believe the reason the test is panicking for you is because this code is not inside a call to [`loom::model`](https://docs.rs/loom/0.2.14/loom/fn.model.html), so there's no loom execution running when the atomics are constructed.

If you replace it with
```rust
#[test]
fn sanity() {
    loom::model(|| {
        let unpark = Box::new(MockUnpark);
        let instant = Instant::now();

        let inner = Inner::new(instant, unpark);

        assert_eq!(inner.elapsed(), 0);
    })
}
```
I think it won't panic any longer.

That said, I don't think this really _needs_ to be a loom test, since there's no concurrency involved. This should be pretty much equivalent to a regular Rust test. Though, if your intention was just to start learning _how_ to write `loom` tests, that makes sense. :)

Reply:
> I believe the reason the test is panicking for you is because this code is not inside a call to loom::model, so there's no loom execution running when the atomics are constructed.

 Hah! D'oh. Makes perfect sense. 

> That said, I don't think this really needs to be a loom test, since there's no concurrency involved. This should be pretty much equivalent to a regular Rust test. Though, if your intention was just to start learning how to write loom tests, that makes sense. :)

Oh, this for sure does not. The ultimate goal is to demonstrate with a loom test that I didn't obviously bungle the work done to `increment`. This 'test' was more just to demonstrate to myself that I could run something simple in a loom rig, which now -- thank you! -- I can. 

I actually have a good sense of how to proceed from here, so this was very helpful.

Reply:
Okay, great! Glad I could help! :)